packages/zoltan2/src/algorithms/partition/Zoltan2_AlgPuLP.hpp

```diff
@@ -252,8 +256,9 @@ class AlgPuLP : public Algorithm<Adapter>
 
   void buildModel(modelFlag_t &flags);
 
-  void scale_weights(size_t n, StridedData<lno_t, scalar_t> &fwgts,
-                     int *iwgts);
+  void scale_weights( size_t n, int nWgt,
```

Comment:
1. This interface does not make clear whether it transfers ownership of `iwgts`.  That lack of clarity could cause memory leaks or double frees.  If you intend to transfer ownership to the callee, then you should pass `iwgts` as `unique_ptr<int[]>`.

2. Why are you passing `ArrayView` by nonconst reference?  That suggests that it's an output parameter.  If so, then why not just return `ArrayView`?

Reply:
I've addressed this in a recent commit. The function was taken as-is from other partitioning methods, but I've updated it within Zoltan2_AlgPuLP to make the interface more clear.